src/components/client/Layout.js

```diff
@@ -1,48 +1,163 @@
 import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import SideNav from './containers/SideNav';
 import M from 'materialize-css';
+import { connect } from 'react-redux';
+import { getSearchProducts } from '../../helpers/api';
+import { saveSearchResult } from '../../actions/example.actions';
+import './Layout.css';
+import logo from './../../images/logo_storeo_white.png';
+
 
 class Layout extends Component {
+  state = {
+    selectedCategoryName: undefined,
+    categoryId: null,
+    queryPhrase: '',
+  }
 
   componentDidMount() {
     M.AutoInit();
+    // this.totalBasketQty();
+  }
+
+  totalBasketQty = () => {
+    let subTotalBasketQty = 0;
+    this.props.basket.forEach(item => {
+      subTotalBasketQty += item.quantity
+    });
+    return subTotalBasketQty;
+  }
+
+  selectCategory = async (categoryId, selectedCategoryName) => {
+    await this.setState({ categoryId, selectedCategoryName });
+  }
```

Comment:
If you have an `async` function that only awaits for a promise you can just return the promise.

```js
selectCategory = (categoryId, selectedCategoryName) => {
  return this.setState({ categoryId, selectedCategoryName });
}
```